tests/graph_components/validators/test_finetuning_validator.py

```diff
@@ -0,0 +1,434 @@
+    domain_with_new_action[KEY_RESPONSES]["utter_new"] = [{"text": "hi"}]
+    domain_with_new_action = Domain.from_dict(domain_with_new_action)
+    importer.get_domain = lambda: domain_with_new_action
+
+    # finetuning
+    loaded_validate = get_validation_method(
+        finetuning=True, load=True, core=core, nlu=nlu
+    )
+    assert importer.get_domain() != old_domain
+    if core:
+        with pytest.raises(InvalidConfigException):
+            loaded_validate(importer=importer, config=config)
+    else:
+        loaded_validate(importer=importer, config=config)
+
+
+def _get_example_config() -> Dict[Text, Any]:
+    return {
+        "language": "en",
+        "pipeline": [
+            {"name": "WhitespaceTokenizer"},
+            {"name": "RegexFeaturizer"},
+            {"name": "LexicalSyntacticFeaturizer"},
+            {"name": "CountVectorsFeaturizer"},
+            {
+                "name": "CountVectorsFeaturizer",
+                "analyzer": "char_wb",
+                "min_ngram": 1,
+                "max_ngram": 4,
+            },
+            {"name": "DIETClassifier", "epochs": 100},
+            {"name": "EntitySynonymMapper"},
+            {"name": "ResponseSelector", "epochs": 100},
+            {
+                "name": "FallbackClassifier",
+                "threshold": 0.3,
+                "ambiguity_threshold": 0.1,
+            },
+        ],
+        "policies": [
+            {"name": "MemoizationPolicy"},
+            {"name": "TEDPolicy", "max_history": 5, "epochs": 100},
+            {"name": "RulePolicy"},
+        ],
+    }
+
+
+def _create_importer_from_config(
+    config: Dict[Text, Any], path: Path, config_file_name: Text
+) -> TrainingDataImporter:
+    config1_path = path / config_file_name
+    rasa.shared.utils.io.write_yaml(config, config1_path, True)
+    return TrainingDataImporter.load_from_config(str(config1_path))
+
+
+@pytest.mark.parametrize("nlu, core", [(True, False), (False, True), (True, True)])
+def test_validate_after_changing_epochs_in_config(
+    tmp_path: Path,
+    get_validation_method: Callable[..., ValidationMethodType],
+    nlu: bool,
+    core: bool,
+):
+    # NOTE: test_model/test_validate_changing_config_epochs (first part)
+
+    # training
+    config1 = _get_example_config()
+    importer = _create_importer_from_config(config1, tmp_path, "config1.yml")
+    validate = get_validation_method(finetuning=False, load=False, nlu=nlu, core=core)
+    validate(importer=importer, config=config1)
+
+    # Change Configuration - replace all epoch settings by a different value
+    config2 = copy.deepcopy(config1)
+    replacements = 0
+    for key in config2:
+        for sub_config in config2[key]:
+            if "epochs" in sub_config:
+                sub_config["epochs"] = sub_config["epochs"] + 5
+                replacements += 1
+    assert (
+        replacements > 0
+    ), 'Please update the test such that the config used here contains "epoch" keys.'
```

Comment:
Do we need to test our tests? 😂

Reply:
kind of :) - I don't see any harm in checking this because the config that is modified here is defined in another function and so you have to know that this might need an update (otherwise it will just silently pass without checking anything)